src/workerd/api/queue.h

```diff
@@ -63,9 +71,198 @@ class WorkerQueue: public jsg::Object {
   uint subrequestChannel;
 };
 
+// Event handler types
+
+// Types for other workers passing messages into and responses out of a queue handler.
+
+struct IncomingQueueMessage {
+  kj::String id;
+  kj::Date timestamp;
+  kj::Array<kj::byte> body;
+  JSG_STRUCT(id, timestamp, body);
+};
+
+struct QueueResponse {
+  uint16_t outcome;
+  bool retryAll;
+  bool ackAll;
+  kj::Array<kj::String> explicitRetries;
+  kj::Array<kj::String> explicitAcks;
+  JSG_STRUCT(outcome, retryAll, ackAll, explicitRetries, explicitAcks);
+};
+
```

Comment:
They do need to be `JSG_STRUCT`s because they're passed across the JS/C++ boundary, but they probably shouldn't be included in type definitions.

I have another PR coming up soon that will lead to `QueueResponse` being returned in response to calling the `queue()` handler on a service binding, but even that doesn't seem to justify getting added to external types if the functionality is protected by an experimental. compatibility flag.